server/src/internalClusterTest/java/org/opensearch/search/SearchCancellationIT.java

```diff
@@ -153,15 +164,55 @@ private SearchResponse ensureSearchWasCancelled(ActionFuture<SearchResponse> sea
             SearchResponse response = searchResponse.actionGet();
             logger.info("Search response {}", response);
             assertNotEquals("At least one shard should have failed", 0, response.getFailedShards());
+            verifyCancellationException(response.getShardFailures());
             return response;
         } catch (SearchPhaseExecutionException ex) {
             logger.info("All shards failed with", ex);
+            verifyCancellationException(ex.shardFailures());
             return null;
         }
     }
 
-    public void testCancellationDuringQueryPhase() throws Exception {
+    private MultiSearchResponse ensureMSearchWasCancelled(ActionFuture<MultiSearchResponse> mSearchResponse,
+        byte childRequestFailureOrder) {
+        MultiSearchResponse response = mSearchResponse.actionGet();
+        byte actualChildRequestFailureOrder = 0;
+        for (int i = 0; i < response.getResponses().length; ++i) {
+            SearchResponse sResponse = response.getResponses()[i].getResponse();
+            // check if response is null means all the shard failed for this search request
+            if (sResponse == null) {
+                Exception ex = response.getResponses()[i].getFailure();
+                assertTrue(ex instanceof SearchPhaseExecutionException);
+                verifyCancellationException(((SearchPhaseExecutionException)ex).shardFailures());
+                // set the ith LSB
+                actualChildRequestFailureOrder |= 1 << i;
+
+            } else if (sResponse.getShardFailures().length > 0) {
+                verifyCancellationException(sResponse.getShardFailures());
+                // set the ith LSB
+                actualChildRequestFailureOrder |= 1 << i;
+            }
```

Comment:
this form of verification using bitwise logic is hard to understand. I would opt for a more simple solution, that is clear to the maintainer when the test fails what actually has gone wrong. 
Perhaps I am just unfamiliar with this being used in tests, but I would chalk it up as a readability issue.